Kickstarter-iOS/Features/Dashboard/Controller/DashboardViewController.swift

```diff
@@ -198,6 +203,43 @@ internal final class DashboardViewController: UITableViewController {
     }
   }
 
+  fileprivate func removeDeprecationWarning() {
+    let dashboardViewController = self.tabBarController?.children.first { viewController in
+      viewController is UIHostingController<DashboardDeprecationView>
+    }
+
+    dashboardViewController?.removeFromParent()
+
+    let dashboardViewControllerView = self.tabBarController?.view.subviews.first(where: { view in
+      view.viewWithTag(-99) != nil
+    })
+
+    dashboardViewControllerView?.removeFromSuperview()
+  }
+
+  fileprivate func showDeprecationWarning() {
+    let dashboardRemovalWarningHostingController = UIHostingController(rootView: DashboardDeprecationView())
+
+    guard let tabController = self.tabBarController as? RootTabBarViewController,
+      let dashboardViewControllerView = dashboardRemovalWarningHostingController.view else { return }
+    dashboardViewControllerView.tag = -99
+    tabController.addChild(dashboardRemovalWarningHostingController)
+    tabController.view.addSubview(dashboardViewControllerView)
+
+    dashboardRemovalWarningHostingController.didMove(toParent: tabController)
+
+    dashboardViewControllerView.translatesAutoresizingMaskIntoConstraints = false
+
+    NSLayoutConstraint
+      .activate([
+        dashboardViewControllerView.leftAnchor.constraint(equalTo: tabController.view.leftAnchor),
+        dashboardViewControllerView.rightAnchor
+          .constraint(equalTo: tabController.view.rightAnchor),
+        dashboardViewControllerView.bottomAnchor.constraint(
+          equalTo: tabController.tabBar.safeAreaLayoutGuide.topAnchor)
+      ])
```

Comment:
I used the `MessageBannerViewController` `MessageBannerViewControllerPresenting` `configureMessageBannerViewController` as a guide here. You'll see a lot of similarities in how the view controller is add as a child and the view is added on top of the `tabController` `view`.

It's the same concept because that toast you might have seen after backing is this `MessageBannerViewController` and it's used everywhere in the app, so it's pretty reliable.